agent/api/task/service_connect.go

```diff
@@ -65,3 +81,246 @@ type DNSConfigEntry struct {
 	HostName string `json:"hostName"`
 	Address  string `json:"address"`
 }
+
+// ParseServiceConnectAttachment parses the service connect container name and service connect config value
+// from the passed attachment.
+func ParseServiceConnectAttachment(scAttachment *ecsacs.Attachment, taskContainers []*ecsacs.Container, networkMode string) (*ServiceConnectConfig, error) {
+	scConfigValue := &ServiceConnectConfig{}
+	containerName := ""
+	for _, property := range scAttachment.AttachmentProperties {
+		switch aws.StringValue(property.Name) {
+		case serviceConnectConfigKey:
+			// extract service connect config value from attachment properties,
+			// and translate ecsacs.Attachment.Value to ServiceConnectConfig
+			data, err := strconv.Unquote(string(*property.Value))
+			if err != nil {
+				return nil, fmt.Errorf("failed to unqote attachment property value which contains service connect config value: %w", err)
+			}
+			if err := json.Unmarshal([]byte(data), scConfigValue); err != nil {
+				return nil, fmt.Errorf("failed to unmarshal attachment property value which contains service connect config value: %w", err)
+			}
+		case serviceConnectContainerNameKey:
+			// extract service connect container name from attachment properties
+			containerName = aws.StringValue(property.Value)
+		default:
+			return nil, fmt.Errorf("received an unrecognized attachment property: %s", property.String())
+		}
+	}
+	scConfigValue.ContainerName = containerName
+	if err := ValidateSCConfig(scConfigValue, taskContainers, networkMode); err != nil {
+		return nil, fmt.Errorf("service connect config validation failed: %w", err)
+	}
+	return scConfigValue, nil
+}
+
+// validateDnsConfig validates the service connnect dns config exists if egress config exists.
+func validateDnsConfig(scDnsConfligList []DNSConfigEntry, scEgressConfig EgressConfig) error {
+	// DNSConfig will also be empty as EgressConfig can be empty for the first service
+	// since there are no other tasks that it can talk to
+	if len(scDnsConfligList) < 1 {
+		if scEgressConfig.ListenerName != "" {
+			return fmt.Errorf("service connect config validation: no service connect DNS config")
+		}
+	}
+	for _, dnsEntry := range scDnsConfligList {
+		// HostName is a required field
+		if dnsEntry.HostName == "" {
+			return fmt.Errorf("service connect config validation: no service connect DNS config host name")
+		}
+		// Address is a required field
+		if dnsEntry.Address == "" {
```

Comment:
Could we add a validation that this is a valid v4/v6 IP?

Reply:
Validation and its test are added in commit 3d8d95406e74fc25d3c00d384a11cd5cd4ae55f9. Please take a look on `service_connect_validation.go` and `service_connect_validation_test.go`. Thanks.